Userland/Libraries/LibWeb/DOM/ARIAMixin.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022, Jonah Shafran <jonahshafran@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/DOM/ARIAMixin.h>
+
+namespace Web::DOM {
+
+// https://www.w3.org/TR/wai-aria-1.2/#introroles
+FlyString ARIAMixin::role_or_default() const
+{
+    // 1. Use the rules of the host language to detect that an element has a role attribute and to identify the attribute value string for it.
+    auto role_string = this->role();
```

Comment:
We generally avoid using `this->` unless it's required because a method name conflicts with a variable name.

```suggestion
    auto role_string = role();
```